apps/nova/src/components/TutorProfile/Ratings.tsx

```diff
@@ -1,22 +1,72 @@
 import { Loading, Timeline, TimelineItem } from "@litespace/luna";
 import { IRating } from "@litespace/types";
 import { UseQueryResult } from "@tanstack/react-query";
-import React, { useMemo } from "react";
+import React, { useMemo, useState, useCallback } from "react";
 import { Star } from "react-feather";
-import Title from "./Title";
-import Rating from "./Rating";
+import Title from "@/components/TutorProfile/Title";
+import Rating from "@/components/TutorProfile/Rating";
+import { useRender } from "@/hooks/render";
+import EditRating from "@/components/TutorProfile/EditRating";
+import DeleteRating from "@/components/TutorProfile/DeleteRating";
 
 const Ratings: React.FC<{
   query: UseQueryResult<IRating.Populated[], Error>;
 }> = ({ query }) => {
+  const [ratingEdited, setRatedEdited] = useState<IRating.Populated | null>(
+    null
+  );
+  const [ratingDeleted, setRatingDeleted] = useState<{
+    ratingId: number;
+    tutorId: number;
+  }>({
+    ratingId: 0,
+    tutorId: 0,
+  });
```

Comment:
It is better to initialize it to null. 

```suggestion
    ratingId: number;
    tutorId: number;
  } | null>(null);
```